feat: Add script to dynamically select docker or podman

pkarakal opened the pull request:
This script is created to check for the container engine the user
is running. After checking for the engine it also checks the cgroup API
they are using. If they are using cgroup v2 and they have docker installed,
the script will automatically switch to podman. If the user has cgroup v1 enabled,
and the docker engine is installed, the script will use docker. Else it will default to
podman. If neither is installed or can be used, it will exit with non 0 code. Podman can
be used with both cgroup API versions as it is daemon-less and run rootless by default.
This script is necessary for developers that aren't running  the docker engine or that
their OS doesn't support the docker engine like some Linux distros (eg Fedora).

pkarakal committed:
feat: Add script to dynamically select docker or podman

This script is created to check for the container engine the user
is running. After checking for the engine it also checks the cgroup API
they are using. If they are using cgroup v2 and they have docker installed,
the script will automatically switch to podman. If the user has cgroup v1 enabled,
and the docker engine is installed, the script will use docker. Else it will default to
podman. If neither is installed or can be used, it will exit with non 0 code. Podman can
be used with both cgroup API versions as it is daemon-less and run rootless by default.
This script is necessary for developers that aren't running  the docker engine or that
their OS doesn't support the docker engine like some Linux distros (eg Fedora).

bnase commented:
@pkarakal Thanks for the effort! Really never heard of podman before :) I ll give it a go. Why would u prefer this over Docker btw? More lightweight?

pkarakal commented:
> @pkarakal Thanks for the effort! Really never heard of podman before :) I ll give it a go. Why would u prefer this over Docker btw? More lightweight?

I am using Fedora and by default it uses cgroup2 API and the current releases of Docker don't support cgroups2 API (though it is planned to be release later this year) and it cannot start containers (though one can always switch the cgroups API by modifying the kernel option [simple as running `grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
` and restarting]). Also podman is daemon-less and supports rootless containers thus making it more secure by default. 
(Podman is also capable of running both containers and pods)

I have been using podman for development and didn't quite find it useful to switch the tool just for the course so I just added a script to switch the OCI tool, and since their API is mostly the same why not? :grin:

pkarakal committed:
fix: Add rshared bind propagation to image volume

When using podman the default bind propagation is rprivate and the
directories are not shared with the container. By adding the rshared,
I am making all the directories recursively shared between host and
container and the user is able to interact with the files on the volume.
